models/src/test/scala/coop/rchain/models/MemoSpec.scala

```diff
@@ -0,0 +1,22 @@
+package coop.rchain.models
+import monix.eval.Coeval
+import org.scalatest.{FlatSpec, Matchers}
+
+class MemoSpec extends FlatSpec with Matchers {
+
+  behavior of "Memo"
+
+  it should "memoize the result" in {
+    var timesExecuted = 0
+    val random = new Memo[Int](Coeval.delay {
+      timesExecuted += 1
+      9
+    })
+    assert(timesExecuted == 0)
+    assert(random.get.value() == 9)
+    assert(timesExecuted == 1)
+    assert(random.get.value() == 9)
+    assert(timesExecuted == 1)
+  }
```

Comment:
How does this test compare with plain `Coeval.memoize` ?

Reply:
Coeval.memoize is not stacksafe. I'd rather not dive into testing that - we've got other tests that will catch it.